source/funkin/play/PlayState.hx

```diff
@@ -1169,7 +1170,16 @@ class PlayState extends MusicBeatSubState
     // Dispatch event to conversation script.
     ScriptEventDispatcher.callEvent(currentConversation, event);
 
-    // TODO: Dispatch event to note scripts
+    // Dispatch event to only the specific note script
+    if (Std.isOfType(event, NoteScriptEvent))
+    {
+      var noteEvent:NoteScriptEvent = cast(event, NoteScriptEvent);
+      NoteKindManager.callEvent(noteEvent.note.noteData.kind, noteEvent);
+    }
+    else // Dispatch event to all note scripts
+    {
+      NoteKindManager.callEventForAll(event);
```

Comment:
I'm pretty sure this block is logic that should be part of the ScriptEventDispatcher.

Reply:
i made it part of NoteKindManager, if thats also okay